### ISSUE TITLE
ImportError: No module named 'IPython.nbformat.current'

### ISSUE THREAD

**Issue by alexpetralia** [spyder-ide/spyder#3176]
## Description of your problem

**What steps will reproduce the problem?**

1. Fresh install of spyder3: `sudo apt-get install spyder3`
2. Run spyder3: `spyder3`

**What is the expected output? What do you see instead?**

Spyder3 should open. It does not.

## Versions and main components

* Python Version: 3.4
* Operating system: Linux

## Traceback

```
alex@linux-box ~> spyder3
/home/alex/.local/lib/python3.4/site-packages/IPython/qt.py:13: ShimWarning: The `IPython.qt` package has been deprecated. You should import from qtconsole instead.
  "You should import from qtconsole instead.", ShimWarning)
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/spyderlib/spyder.py", line 2354, in main
    mainwindow = run_spyder(app, options, args)
  File "/usr/lib/python3/dist-packages/spyderlib/spyder.py", line 2242, in run_spyder
    main.setup()
  File "/usr/lib/python3/dist-packages/spyderlib/spyder.py", line 775, in setup
    message=_("Spyder Internal Console\n\n"
  File "/usr/lib/python3/dist-packages/spyderlib/plugins/console.py", line 74, in __init__
    self.find_widget.set_editor(self.shell)
  File "/usr/lib/python3/dist-packages/spyderlib/widgets/findreplace.py", line 264, in set_editor
    from spyderlib.widgets.sourcecode.codeeditor import CodeEditor
  File "/usr/lib/python3/dist-packages/spyderlib/widgets/sourcecode/codeeditor.py", line 57, in <module>
    import IPython.nbformat.current  # in IPython 0.13.2, current is not loaded
ImportError: No module named 'IPython.nbformat.current'
```

Running `spyder` (ie. 2.7) works fine.

**Comment by jitseniesen** [comment]
It looks like you have a version of IPython installed in `/home/alex/.local` ; see the first line of the error message. This is used instead of the IPython installed by `apt-get` and may be the cause of the issue. The quickest solution may be to remove the IPython package in `/home/alex/.local` so that the IPython package in your distribution is used. Presumably, spyder3 works as long as you stick to the packages in your distribution.

My guess is that the IPython package being used is too new for the Spyder that you use. Specifically, it may be issue #2250 which is fixed by 270fc44. If you want to keep the IPython package in `/home/alex/.local` then try installing the beta version of Spyder; see https://github.com/spyder-ide/spyder/releases/tag/v3.0.0b2 .

If this does not help you, please tell us your distribution (Ubuntu, Debian, ...), the version of your spyder3 package, the version of the IPython package in `home/alex/.local` and where you got it from.

**Comment by alexpetralia** [comment]
@jitseniesen Thank you! Following your advice, here is what I did (for anyone else): `cd ~/.local/lib/python3.4/site-packages; mv IPython IPython2` and then restarted Spyder3 - success! Thanks again for your help.
